deploy/efs-csi/01-daemonset.yaml

```diff
@@ -0,0 +1,99 @@
+# Source: https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/51d19a433dcfc47fbb7b7a0e1c8ff6ab98ce87e9/deploy/kubernetes/base/node.yaml
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: efs-csi-node
+  namespace: kube-system
```

Comment:
is `kube-system` the right NS for this?

Reply:
This is a straight copy from the source (L1) (except for the nodeSelector on L17) and is how we tested. If `kube-system` isn't right, what would it be?

Reply:
Probably a new NS for just this purpose. I would be reluctant to put things in kube-system that don't need to be there. `openshift-csi-drivers` maybe?

Reply:
Agree, we should create a new namespace and use that if possible.  Any changes to an OCP provided namespace should only be done if we have no other options.

Reply:
`openshift-csi-drivers` makes sense to me :+1:

Reply:
+1 if customer doesn't administer the namespace

Reply:
I tried this and it makes the pod fail to start:

```
  Warning  FailedMount  2m37s                 kubelet, ip-10-0-139-172.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[efs-mnt], unattached volumes=[default-token-6mjb2 efs-mnt]: timed out waiting for the condition
  Warning  FailedMount  30s (x10 over 4m40s)  kubelet, ip-10-0-139-172.ec2.internal  MountVolume.MountDevice failed for volume "efs-pv" : kubernetes.io/csi: attacher.MountDevice failed to create newCsiDriverClient: driver name efs.csi.aws.com not found in the list of registered CSI drivers
```

What now?

Reply:
I put this back for now so we can make forward progress.

Reply:
Dug a bit more here.

The daemonset fails to create in a generic namespace with
```Error creating: pods "efs-csi-node-" is forbidden: pods with system-node-critical priorityClass is not permitted in efsap namespace```

It needs to be in the `system-node-critical` priority class because otherwise:
```
Error creating: pods "efs-csi-node-" is forbidden: unable to validate against any security context constraint: [
  provider restricted:
    .spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used
    spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used
    <same for spec.volumes[1, 2, 3]>
    spec.containers[0].securityContext.privileged: Invalid value: true: Privileged containers are not allowed
    spec.containers[0].securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used
    spec.containers[0].securityContext.containers[0].hostPort: Invalid value: 9809: Host ports are not allowed to be used
    <same for spec.containers[1, 2]>
]
```

So I gather that we would need to create some kind of SCC/(Cluster)Role(Binding) etc. and attach it to our bespoke namespace. I doddled around with a couple of things (like cloning `kube-system`'s annotations) to no avail. Without a deeper understanding of the RBAC-isms, I'm shooting in the dark.

Reply:
@drow and I spent some time working through this and we think we have a path forward. It entails:
- Creating our namespace, let's call it `openshift-efs-csi`.
- Creating a ServiceAccount in that namespace, let's call it `efs-csi-sa`
- Creating a SCC that
  - allows at least*:
    - privileged containers
    - hostNetwork and hostPorts
    - volumes.hostPath
  - includes user `system:serviceaccount:openshift-efs-csi:efs-csi-sa` (that last bit is `$namespace:$serviceaccount`)
- In the DaemonSet:
  - Using `metadata.namespace: openshift-efs-csi`
  - In `spec.template.spec`:
     - Removing `priorityClassName`
     - Adding `serviceAccount: efs-csi-sa`

I'll run through it a bit more to make sure, then update the PR accordingly.

This shouldn't require changes to the customer or SRE instructions.

*NB: when experimenting, we got this to work by cloning the `privileged` SCC. It is unclear whether we can reduce from there and still have stuff work.

Reply:
Added a commit to the PR for this.

Reply:
This is clean to me, we are not modifying any default SCC's and we have a logically separate namespace for the ESI driver, I'm +1 good work @2uasimojo and @dustman9000 

@jewzaam @cblecker could I get a +1 from you, we'd like to get this merged today.